concordia/forms.py

```diff
@@ -56,11 +68,15 @@ def save(self, commit=True):
             setattr(instance, role_dict[role], 1)
         if commit:
             instance.save()
+
+        if "newsletterOptIn" in self.data and self.data["newsletterOptIn"]:
+            # Add this user to the Newsletter group if they opted in.
+            post_save.connect(add_user_to_newsletter_group, sender=User)
```

Comment:
I'm a little confused by the ordering here — since we're just calling save on the line before, why not just add the user directly?

Also, do we have a process for opt-ing out?

Reply:
Yeah, I thought that would work too. I guess there's a bit of weirdness with how Django clears and refreshes many-to-many relationships. At this point the id for newly-created Users doesn't exist, although it seems like it should.
GovDelivery manages the newsletter - I think the opt-out thing is handled through that.

Reply:
Users have to explicitly opt-in to the newsletter, as per wider Library policy. They can opt out through gov delivery emails or going to the gov delivery site.